### PR TITLE
fix: add an info message, and converts info to debug on non-pem files

### LINKED ISSUE #26929: Ignore unrecognized truststore formats if `--truststore-paths` is a directory
### Description

When the system truststore is specified as a directory, we should ignore any files that are not recognized as a valid truststore. It is a valid use case (e.g. for K8s deployments) that a directory could include more than just truststores.

### Discussion

_No response_

### Motivation

https://github.com/keycloak/keycloak/pull/26912#pullrequestreview-1872080857

### Details

_No response_

### PR DESCRIPTION
**Pull request by shawkins**
closes: #26929

### CODE REVIEW

File: services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java [hunk 1]
```diff
@@ -184,7 +196,7 @@ static KeyStore loadStore(String path, String type, char[] password) {
         }
     }
 
-    private static void mergePemFile(KeyStore truststore, String file, boolean isPem) {
+    private static boolean mergePemFile(KeyStore truststore, String file, boolean isPem, List<String> loadedFiles) {
```
Comment (vmuzikar) [inline services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java:199]: `loadedFiles` doesn't seem to be used?

File: services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java [hunk 2]
```diff
@@ -85,20 +87,30 @@ static KeyStore createMergedTruststore(String[] truststores, boolean trustStoreI
             includeDefaultTruststore(truststore);
         }
 
-        mergeFiles(truststores, truststore, true);
+        List<String> loadedFiles = new ArrayList<>();
```
Comment (vmuzikar) [inline services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java:90]: Just a nitpick. I'd suggest using a more clear name, maybe `loadedFilesFromDir`?

**Review by vmuzikar — COMMENTED** [review]
@shawkins Thanks for the PR!

File: services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java (hunk 2 above)
Comment (shawkins) [reply services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java]: No problem, updated the name.

File: services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java (hunk 1 above)
Comment (shawkins) [reply services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java]: It isn't, I had initially added the files in that method, then moved it out for consitency - but didn't remove the parameter. It's gone now.

File: services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java [hunk 3]
```diff
@@ -85,20 +87,30 @@ static KeyStore createMergedTruststore(String[] truststores, boolean trustStoreI
...
+                mergeFiles(Stream.of(f.listFiles()).map(File::getAbsolutePath).toArray(String[]::new), truststore, false, discoveredFiles);
             } else {
                 if (file.endsWith(".p12") || file.endsWith(".pfx")) {
                     mergeTrustStore(truststore, file, loadStore(file, PKCS12, null));
+                    if (!topLevel) {
+                        discoveredFiles.add(f.getAbsolutePath());
+                    }
                 } else {
-                    mergePemFile(truststore, file, topLevel);
+                    if (mergePemFile(truststore, file, topLevel) && topLevel) {
```
Comment (vmuzikar) [inline services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java:111]: One more thing. Should the condition be instead like this?
```suggestion
                    if (mergePemFile(truststore, file, topLevel) && !topLevel) {
```

**Comment by vmuzikar** [comment]
Can you please also rebase?

**Comment by shawkins** [comment]
> Can you please also rebase?

Done

File: services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java (hunk 3 above)
Comment (shawkins) [reply services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java]: Yes, it should be.

File: services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java [hunk 4]
```diff
@@ -84,20 +86,30 @@ static KeyStore createMergedTruststore(String[] truststores, boolean trustStoreI
...
             } else {
                 if (file.endsWith(".p12") || file.endsWith(".pfx")) {
                     mergeTrustStore(truststore, file, loadStore(file, PKCS12, null));
+                    if (!topLevel) {
+                        discoveredFiles.add(f.getAbsolutePath());
+                    }
                 } else {
-                    mergePemFile(truststore, file, topLevel);
+                    if (mergePemFile(truststore, file, topLevel) && !topLevel) {
+                        discoveredFiles.add(file);
```
Comment (vmuzikar) [inline services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java:111]: I'm sorry, missed this. Last comment, I promise. ;)
```suggestion
                        discoveredFiles.add(f.getAbsolutePath());
```
to align with pkcs12 above.

Comment (shawkins) [reply services/src/main/java/org/keycloak/truststore/TruststoreBuilder.java:111]: Sure - that's the elusive consistency I was shooting for :)

**Review by vmuzikar — APPROVED** [review]

**Pull request merged**